types/validator_set.go

```diff
@@ -966,6 +984,21 @@ func ValidatorSetFromProto(vp *tmproto.ValidatorSet) (*ValidatorSet, error) {
 	return vals, vals.ValidateBasic()
 }
 
+// ValidatorSetFromExistingValidators takes an existing array of validators and rebuilds
+// the exact same validator set that corresponds to it without changing the proposer priority or power
+func ValidatorSetFromExistingValidators(valz []*Validator) (*ValidatorSet, error) {
```

Comment:
can't we combine this func with NewValidatorSet?

Reply:
in what way? have a check to see if the proposer priority is not nil for the validators? and if so then follow the logic of this function instead

Reply:
I think my reason for having a separate function is that I wanted to show the clear distinction between the concept of making a validator set from new validators and making a validator set from existing validators. I can change this if you prefer going for simplicity

Reply:
> have a check to see if the proposer priority is not nil for the validators? and if so then follow the logic of this function instead

yeah

Reply:
Ok, I can make the change in another PR